### PR TITLE
Simplify InputStreamFileSystem and InputStreamFileHandle implementations

### PR DESCRIPTION
**Pull request by majetideepak**
The current InputStreamFileSystem class uses a `streams_` map, which introduces additional latency as each read call touches it.
This PR removes the need for this map. This simplifies both InputStreamFileSystem and InputStreamFileHandle implementations.
I ran a test with TPCH Q1, SF 10, single worker and observed an improvement of 15% CPU time for Parquet format.

The next steps would be to extend DuckDB to directly support BufferedInput stream from the DWRF world.

### CODE REVIEW

File: velox/dwio/parquet/reader/ParquetReader.h [hunk 1]
```diff
@@ -71,14 +71,10 @@ class ParquetReader : public dwio::common::Reader {
       const dwio::common::RowReaderOptions& options = {}) const override;
 
  private:
-  duckdb::InputStreamFileSystem* getFileSystem() {
```
Comment (majetideepak) [inline velox/dwio/parquet/reader/ParquetReader.h:74]: Changing this from a singleton to an instance simplifies the whole thing.

File: velox/dwio/parquet/reader/ParquetReader.h [hunk 2]
```diff
@@ -71,14 +71,10 @@ class ParquetReader : public dwio::common::Reader {
...
-    static duckdb::InputStreamFileSystem fileSystem;
```
Comment (9prady9) [inline velox/dwio/parquet/reader/ParquetReader.h:75]: Is the behaviour of earlier constructor based `fileSystem` variable and the one we have now via `fileSystem_ =
      std::make_unique<duckdb::InputStreamFileSystem>(std::move(stream))` identical ?
      
Unique_ptr are definitely better to use than passing around raw pointers I think.

File: velox/dwio/parquet/reader/duckdb/InputStreamFileHandle.h [hunk 3]
```diff
@@ -22,33 +22,16 @@ namespace facebook::velox::duckdb {
 
 class InputStreamFileSystem;
 
+// Implements the DuckDB FileHandle API
 class InputStreamFileHandle : public ::duckdb::FileHandle {
```
Comment (9prady9) [inline velox/dwio/parquet/reader/duckdb/InputStreamFileHandle.h:26]: Isn't this essentially `using InputStreamFileHandle = ::duckdb::Filehandle` now ? Or is a separate inherited class somehow mandatory to use that class ?

File: velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h [hunk 4]
```diff
@@ -39,152 +45,101 @@ class InputStreamFileSystem : public ::duckdb::FileSystem {
...
   }
 
  private:
-  std::atomic<int> nextStreamId_ = 1;
-  // Maps stream ID to file handle counter and input stream.
-  folly::Synchronized<folly::F14FastMap<
-      int,
-      std::pair<int, std::unique_ptr<dwio::common::InputStream>>>>
-      streams_;
+  std::unique_ptr<dwio::common::InputStream> stream_;
```
Comment (9prady9) [inline velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:142]: Earlier, this object used to manage multiples streams/file-io. Now it doesn't ? Is that understanding correct ?

**Review by 9prady9 — COMMENTED** [review]
I have posted some curious questions, thank you for letting me know about the change @majetideepak

File: velox/dwio/parquet/reader/ParquetReader.h (hunk 2 above)
Comment (majetideepak) [reply velox/dwio/parquet/reader/ParquetReader.h:75]: The behavior is the same as before. But it changed from being a singleton before to each ParquetReader instance having its own `duckdb::InputStreamFileSystem` instance.

File: velox/dwio/parquet/reader/duckdb/InputStreamFileHandle.h (hunk 3 above)
Comment (majetideepak) [reply velox/dwio/parquet/reader/duckdb/InputStreamFileHandle.h:26]: This and `duckdb::InputStreamFileSystem` both are needed to satisfy the DuckDB Parquet Reader API requirements.

File: velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h [hunk 5]
```diff
@@ -26,11 +26,17 @@
...
+// Implements the DuckDB FileSystem API on top of dwio::common::InputStream.
+// Hence, an instance always supports only a specific path and limited API.
+// This class owns the InputStream instance passed from the HiveConnector.
+// TODO: Work with DuckDB to directly support a InputStream API.
 class InputStreamFileSystem : public ::duckdb::FileSystem {
  public:
-  InputStreamFileSystem() {}
+  InputStreamFileSystem(std::unique_ptr<dwio::common::InputStream> stream)
+      : stream_(std::move(stream)) {}
   ~InputStreamFileSystem() override = default;
```
Comment (mbasmanova) [inline velox/dwio/parquet/reader/duckdb/InputStreamFileSystem.h:37]: nit: add an empty line before this method

**Review by mbasmanova — APPROVED** [review]
@majetideepak Looks good to me.

**Pull request closed without merging**